java/8/Dockerfile

```diff
@@ -1,7 +1,84 @@
+# docker build -t imqs/java-base:8-test --build-arg Maven_Nexus_User=$Maven_Nexus_User --build-arg Maven_Releases_Deployment=$Maven_Releases_Deployment --build-arg Maven_Snapshots_Deployment=$Maven_Snapshots_Deployment --build-arg Maven_Github_gerhardvanwyk=$Maven_Github_gerhardvanwyk .
+
 FROM imqs/ubuntu-base
 
-RUN apt-get update \
-	&& apt-get install openjdk-8-jre -y
+RUN apt update && \
+	apt install -y \
+		openjdk-8-jdk \
+		wget && \
+	rm -rf /var/lib/apt/lists/*
+
+RUN wget https://www-us.apache.org/dist/maven/maven-3/3.5.4/binaries/apache-maven-3.5.4-bin.tar.gz -P /tmp && \
+	tar xf /tmp/apache-maven-*.tar.gz -C /opt && rm /tmp/apache-maven-*.tar.gz && \
+	mv /opt/apache-maven-* /opt/maven
+ENV MAVEN_HOME=/opt/maven
+ENV PATH=${MAVEN_HOME}/bin:${PATH}
+
+ARG Maven_Nexus_User
+ARG Maven_Releases_Deployment
+ARG Maven_Snapshots_Deployment
+ARG Maven_Github_gerhardvanwyk
+RUN mkdir -p /root/.m2
+RUN echo "<settings> \

```

Comment:
Again I don't think we need to be able to build in this container. Having said that:
So every image and therefore every project will have its own maven repository. This means that we will have a massive amount of duplication of the many shared libraries. It also means that it becomes awkward if you want to build locally using chnages pushed to your local .m2 folder?

Maybe I'm missing what you are trying to do?

We took this approach on our side: https://github.com/IMQS/imqs-wip-service/blob/team-asset-containers/build.sh

Reply:
Hmmmm... so two parts

1. What I was trying to do with including the settings file in the base images is affectingly addressing the issues that the build.sh is bringing up in the start of the file. This keeps everything contained and self sustained. There is no dependency on the machine building the jar because everything is all ready setup inside the container.

2. Crap... making changes in your local .m2 folder... thats a good point... as rare as this case may be, It will still be very annoying the day that someone wants to do something like this... The only solution I can think of is doing the same thing that we do for maps, and that is to build the base image with your changes your self. I could add a second dockerfile that copys your local .m2 into the container to streamline this process?